java-api-core/src/com/kinvey/java/store/BaseDataStore.java

```diff
@@ -409,7 +410,8 @@ public void pushBlocking() throws IOException {
      * should be used with {@link StoreType#SYNC}
      * @param query query to pull the objects
      */
-    public KinveyPullResponse pullBlocking(Query query) throws IOException {
+    @Nonnull
+    public KinveyPullResponse pullBlocking(@Nullable Query query) throws IOException {
         Preconditions.checkArgument(storeType != StoreType.NETWORK, "InvalidDataStoreType");
```

Comment:
If we have another `pullBlocking` method that does not take a `Query`, why is the `Query` parameter allowed to be `null`?

Reply:
All `pullBclocking` methods have a `Query` in the parameters.